AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/SignOut/InitiateSignOut.swift

```diff
@@ -19,7 +19,11 @@ struct InitiateSignOut: Action {
         logVerbose("\(#fileID) Starting execution", environment: environment)
 
         let event: SignOutEvent
-        if signOutEventData.globalSignOut {
+        if case .hostedUI(let options) = signedInData.signInMethod,
+           options.preferPrivateSession == false {
+            event = SignOutEvent(eventType: .invokeHostedUISignOut(signOutEventData,
+                                                                   signedInData))
+        } else if signOutEventData.globalSignOut {
```

Comment:
what exactly would be the logic when `options.preferPrivateSession` is `true`?

Reply:
Will be handled in the next PR - https://github.com/aws-amplify/amplify-ios/pull/2030